test/fluree/db/ledger/docs/identity/signatures_test.clj

```diff
@@ -0,0 +1,204 @@
+      json/parse))
+
+
+;; Create auth records -> root, all persons, all persons no handles
+(defn create-auths [ledger]
+  (let [keys     (vec (repeatedly 3 fdb/new-private-key))
+        add-auth [{:_id   "_auth"
+                   :id    (get-in keys [0 :id])
+                   :roles [["_role/id" "root"]]}
+                  {:_id   "_auth"
+                   :id    (get-in keys [1 :id])
+                   :roles ["_role$allPersons"]}
+                  {:_id   "_auth"
+                   :id    (get-in keys [2 :id])
+                   :roles ["_role$noHandles"]}
+                  {:_id   "_role$allPersons"
+                   :id    "allPersons"
+                   :rules ["_rule$allPersons"]}
+                  {:_id   "_role$noHandles"
+                   :id    "noHandles"
+                   :rules ["_rule$allPersons" "_rule$noHandles"]}
+                  {:_id               "_rule$allPersons"
+                   :id                "role$allPersons"
+                   :collection        "person"
+                   :collectionDefault true
+                   :fns               [["_fn/name" "true"]]
+                   :ops               ["all"]}
+                  {:_id        "_rule$noHandles"
+                   :id         "noHandles"
+                   :collection "person"
+                   :predicates ["person/handle"]
+                   :fns        [["_fn/name" "false"]]
+                   :ops        ["all"]}]]
+    [keys (->> add-auth
+               (fdb/transact-async (:conn test/system) ledger)
+               <!!)]))
+
+
+(deftest create-auth-test
+  (let [ledger (test/rand-ledger test/ledger-chat)
+        [_ {:keys [status flakes]}] (create-auths ledger)]
+    (is (= 200 status))
+
+    (is (= 28 (count flakes)))))
+
+
+;; Endpoint test - unsigned queries should throw an error, no signature
+
+(deftest unsigned-queries-test
+  (testing "An unsigned query should throw an error"
+    (let [ledger       (test/rand-ledger test/ledger-chat)
+          endpoint-map {"query"       {:select ["*"] :from "person"}
+                        "block"       {:block 1}
+                        "graphql"     {:query "{  graph {  _collection (sort: {predicate: \"name\", order: ASC}) { _id name spec version doc}}}"}
+                        "sparql"      "SELECT ?name \nWHERE \n {\n ?collection fd:_collection/name ?name. \n}"
+                        "history"     {:history [["_collection/name" "_user"]]}
+                        "multi-query" {:person {:select ["*"] :from "person"}}}
+          opts         {:throw-exceptions false}
+          resps        (reduce-kv (fn [acc key val]
+                                    (conj acc (send-parse-request ledger key val opts)))
+                                  [] endpoint-map)]
+
+      (is (every? #(= 401 (:status %)) resps))
+
+      (is (every? #(= "Request requires authentication." (:message %)) resps)))))
+
+
+;; Endpoint test - /query - should return filtered results
```

Comment:
I think this comment would be better served as a `testing` form inside of `signed-queries-test`